From 45f0211f05179347c5b395944dcdd64efa563c47 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 20 Feb 2023 11:41:33 +0200
Subject: [PATCH] [SW-123737] gaudi2: ext_rom: set privileged mode to ARCs

This patch adds support to set ARCs to privileged mode as
part of erom init. However till that point ARCs mode is
changed to user mode due to reset and if it tries to
access registers which are set to privilege/secure mode
and not been reset as part of reset the arc gets stuck.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ia8d966a3c9f841f71ae44275960080a71e0a76c7
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index 2c6250b..8c2f783 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -96,6 +96,7 @@
 	 * so the udelay init is done once before we start.
 	 */
 	cpu_boot_if_init();
+	ppboot_secure_arc();
 	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_PREBOOT);
 	/* Enable the errors communication */
 	cpu_boot_err_set(CPU_BOOT_ERR_ENABLED);
